src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -286,9 +291,13 @@ private function createDelayQueue(int $delay, ?string $routingKey)
         return $queue;
     }
 
-    private function getRoutingKeyForDelay(int $delay): string
+    private function getRoutingKeyForDelay(int $delay, ?string $finalRoutingKey): string
     {
-        return str_replace('%delay%', $delay, $this->connectionOptions['delay']['routing_key_pattern']);
+        return str_replace(
+            ['%delay%', '%routing_key%'],
+            [$delay, $finalRoutingKey ?: 'no_routing_key'],
```

Comment:
This breaks again if somebody uses `no_routing_key` as routing key. I know it's unlikely but solvable by using an empty string as default.